src/puya/build-compilation-set-mapping.ts

```diff
@@ -17,7 +17,7 @@ export function buildCompilationSetMapping({
   return awst.reduce((acc, cur) => {
     if (setIds.has(cur.id.toString())) {
       const matchedPath = inputPaths.find((p) => p.sourceFile === cur.sourceLocation.file)
-      if (matchedPath) {
+      if (matchedPath?.outDir) {
```

Comment:
Is this a behaviour change?

Reply:
@neilcampbell good point, i moved the check inside - its the `mkDirIfNotExists` that was causing trouble when dry run mode passes objects with outDir set to empty string.